# Simulation mode
simulation_mode: false             # Is set in launch file!
gazebo_simulation: true
sync_mode: true

auto_enable_plan: true

robot: carla
### Kinematic configuration parameter

# Debug mode
activate_output: true
debug_info_lmpcc: false
activate_controller_node_output: true
plotting_result: true

# Clock frequency //hz
clock_frequency: 10

#Optimal control problem dimensions
state_dim: 6
control_dim: 3

# Parameters for collision constraints
obstacles:
      max_obstacles: 6
      n_discs: 3
      ego_l: 4.540
      ego_w: 2.250
      
delays:
    input_delay: 0  # In timesteps (0.1s)
      
# Kinematic chain parameter
robot_base_link: base_link
target_frame: map # world fixed frame but will be changed according to robot_state_topic

# Weight on the slack variable for collision collision
slack_weight: 10000
repulsive_weight: 0.1
reference_velocity: 5.0
min_velocity: 1.0
initial_speed: 0.0

#Road parameters
road_width_right: 2.1
road_width_left: 4.7

scenarios:
      enable_static_obstacles: false
      multithread: true
      polygon:
        range: 3 #1
        checked_constraints: 150 # Should be scaled with scenarios
        inner_approximation: 6
      iterative_approach: 
        enabled: false
        max_iterations: 2
      database:
        truncated: true
        truncated_radius: 3.5 # in terms of sigma (only used if truncated gaussian)
        build_database: false # ONLY CHANGE IF NO DATABASE
        size: 10
      epsilon: 0.6
      beta: 1e-6
      batch_count: 50 # doesnt work atm?
      seed: -1 #1592207839 # -1 is random!
      #active_constraints: 2
      #removal_count: 50
      #sample_count: 3950 #1250 #3950
      removal_count: 50 # 3sigma: 50  # 1 sigma 200
      sample_count: 53050 # 3sigma: 53050  # 1 sigma 820
      visualisation:
          discs_to_draw: [0, 1, 0]
          indices_to_draw: [0, 6, 12, 19] 
          #[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
          #[0, 8, 16, 24, 29] #[20, 21, 22, 23, 24, 25, 26, 27, 28, 29]
          #[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 ,16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29]
          all_scenarios: true # Scenarios may remain for a while if the current batch is small
          selected_scenarios: false # Is not accurate -> needs more info from polygon constructor
          removed_scenarios: false
          ellipsoids: false
          constraints: true

vru:
      radius: 0.4 #0.75
      eps_t: 0.9
